src/routes/transactions/transactions.service.ts

```diff
@@ -175,4 +185,119 @@ export class TransactionsService {
       results,
     };
   }
+
+  private getDayInMillis(timestamp: number, timezoneOffset?: string): number {
+    const date = new Date(timestamp);
+    if (timezoneOffset !== undefined) {
+      date.setUTCSeconds(parseInt(timezoneOffset) || 0);
+    }
+    return Date.UTC(date.getFullYear(), date.getMonth(), date.getDate());
+  }
+
+  private groupByDay(
+    transactions: TransactionHistory[],
+    timezoneOffset?: string,
+  ): TransactionHistoryGroup[] {
+    return Object.entries(
+      groupBy(transactions, (transaction) =>
+        this.getDayInMillis(transaction.transaction.timestamp, timezoneOffset),
+      ),
+    ).map(
+      ([timestamp, transactions]) =>
+        <TransactionHistoryGroup>{
+          timestamp: Number(timestamp),
+          transactions: transactions,
+        },
+    );
+  }
+
+  private adjustTransactionHistoryPage(
+    limit?: number,
+    offset?: number,
+  ): PaginationData {
+    if (offset !== undefined && offset > 0) {
+      if (limit === undefined) {
+        limit = 20; //default limit
+      }
+      return new PaginationData(limit + 1, offset - 1);
+    }
+    return new PaginationData(limit, offset);
+  }
```

Comment:
You always have a default in the route. No need to set the defaults here.

Reply:
@fmrsabino default pagination is coming here https://github.com/5afe/safe-client-gateway-nest/pull/263 if you want to take a look.

Reply:
> @fmrsabino default pagination is coming here #263 if you want to take a look.

You already have a default on the route for `PaginationData`. https://github.com/5afe/safe-client-gateway-nest/blob/main/src/routes/common/decorators/pagination.data.decorator.ts